### PR TITLE
Taskbar: Scale window icon bitmap if it's not 16x16

### PR DESCRIPTION
**Pull request by luk1337**
Fixes: #5806

### CODE REVIEW

File: Userland/Services/Taskbar/TaskbarWindow.cpp [hunk 1]
```diff
@@ -297,8 +297,17 @@ void TaskbarWindow::wm_event(GUI::WMEvent& event)
     case GUI::Event::WM_WindowIconBitmapChanged: {
         auto& changed_event = static_cast<GUI::WMWindowIconBitmapChangedEvent&>(event);
         if (auto* window = WindowList::the().window(identifier)) {
-            if (window->button())
-                window->button()->set_icon(changed_event.bitmap());
+            if (window->button()) {
+                auto icon = changed_event.bitmap();
+                if (icon->height() > taskbar_icon_size() || icon->width() > taskbar_icon_size()) {
```
Comment (luk1337) [inline Userland/Services/Taskbar/TaskbarWindow.cpp:302]: maybe `!=` instead of `>`?

**Comment by GMTA** [comment]
With screen scaling != 1, wouldn't you need higher res icons than 16x16?

**Comment by luk1337** [comment]
> With screen scaling != 1, wouldn't you need higher res icons than 16x16?

Technically, yes but not with how `2x` scaling works. If I scale the bitmap to 32x32 it looks exactly like it'd look like on `1x`.

File: Userland/Services/Taskbar/TaskbarWindow.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Services/Taskbar/TaskbarWindow.cpp:302]: I think that's sensible, yes.

Comment (luk1337) [reply Userland/Services/Taskbar/TaskbarWindow.cpp]: cool, amended the change.

**Pull request merged**